src/network/networkd.c

```diff
@@ -20,64 +20,33 @@
 #include "sd-daemon.h"
 #include "sd-event.h"
 
-#include "capability-util.h"
 #include "networkd-conf.h"
 #include "networkd-manager.h"
 #include "signal-util.h"
-#include "user-util.h"
 
 int main(int argc, char *argv[]) {
         sd_event *event = NULL;
         _cleanup_manager_free_ Manager *m = NULL;
-        const char *user = "systemd-network";
-        uid_t uid;
-        gid_t gid;
         int r;
 
         log_set_target(LOG_TARGET_AUTO);
         log_parse_environment();
         log_open();
 
-        umask(0022);
-
         if (argc != 1) {
                 log_error("This program takes no arguments.");
                 r = -EINVAL;
                 goto out;
         }
 
-        r = get_user_creds(&user, &uid, &gid, NULL, NULL);
-        if (r < 0) {
-                log_error_errno(r, "Cannot resolve user name %s: %m", user);
+        if (geteuid() == 0) {
+                log_error("This program must be run under an unprivileged user "
+                          "with the following capabilities: CAP_NET_ADMIN "
+                          "CAP_NET_BIND_SERVICE CAP_NET_BROADCAST CAP_NET_RAW");
+                r = -EINVAL;
```

Comment:
In that case, still we can drop privileges by `drop_privileges()`. I think it is not necessary to quit the program.